src/functionaltests/java/com/ericsson/ei/query/FreestyleQueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,149 @@
+package com.ericsson.ei.query;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.commons.io.FileUtils;
+import org.junit.Ignore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+
+import com.ericsson.ei.mongodbhandler.MongoDBHandler;
+import com.ericsson.ei.utils.FunctionalTestBase;
+
+
+@Ignore
+@AutoConfigureMockMvc
+public class FreestyleQueryAggregatedObjectsTestSteps extends FunctionalTestBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreestyleQueryAggregatedObjectsTestSteps.class);
+    
+    private static final String aggregatedObjJsonPath = "src/test/resources/AggregatedDocument.json";
+
+    @Autowired
+    private MockMvc mockMvc;
+    private MvcResult mvcResult;
+    
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+    
+    @Value("${spring.data.mongodb.database}")
+    private String eiDatabaseName;
+    
+    @Value("${aggregated.collection.name}")
+    private String aggrCollectionName;
+    
+    @Value("${missedNotificationCollectionName}")
+    private String missedNotificationCollectionName;
+    
+    @Value("${missedNotificationDataBaseName}")
+    private String missedNotificationDatabaseName;
+
+    private String aggrObj;
+    
+    private ObjectMapper objMapper;
+    
+    public FreestyleQueryAggregatedObjectsTestSteps() {
+        objMapper = new ObjectMapper();
+
+        try {
+            aggrObj = FileUtils.readFileToString(new File(aggregatedObjJsonPath), "UTF-8");
+
+        } catch (IOException e) {
+            LOGGER.error("Failed to open test json files for test. Error message\n: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    @Given("^Aggregated object2 is created$")
+    public void aggregated_object_is_created() throws Throwable {
+        LOGGER.debug("Creating aggregated object in MongoDb");
+        assertEquals(true, createDocumentInMongoDb(eiDatabaseName, aggrCollectionName, aggrObj));
+    }
+
+    @Then("^Perform valid freestyle query on newly created Aggregated object$")
```

Comment:
add this step to the other steps file. Too much overhead to have a runner only for the steps here. This new runner will slow down testing since setting up embedded mongo db takes too much time.

Reply:
I added freestyle queri to same test scenario as the other query test steps.
Should I create a new scenario with same tests steps?